cfy_manager/components/restservice/restservice.py

```diff
@@ -298,35 +297,91 @@ def _validate_admin_password_and_security_config(self):
+            # stats,FRONTEND,,,1,1,2000,7,553,83778,0,0,0,,,,,OPEN,,,,,,,,,1,1,0,,,,0,1,0,1,,,,0,6,0,0,0,0,,1,1,7,,,0,0,0,0,,,,,,,,  # noqa
+            # stats,BACKEND,0,0,0,0,200,0,553,83778,0,0,,0,0,0,0,UP,0,0,0,,0,89,0,,1,1,0,,0,,1,0,,0,,,,0,0,0,0,0,0,,,,,0,0,0,0,0,0,0,,,0,0,0,0,  # noqa
+            # postgres,FRONTEND,,,0,0,2000,0,0,0,0,0,0,,,,,OPEN,,,,,,,,,1,2,0,,,,0,0,0,0,,,,,,,,,,,0,0,0,,,0,0,0,0,,,,,,,,  # noqa
+            # postgres,postgresql_192.0.2.46_5432,0,0,0,0,100,0,0,0,,0,,0,0,0,0,DOWN,1,1,0,1,1,89,89,,1,2,1,,0,,2,0,,0,L7STS,503,3,,,,,,,0,,,,0,0,,,,,-1,HTTP status check returned code <503>,,0,0,0,0,  # noqa
+            # postgres,postgresql_192.0.2.47_5432,0,0,0,0,100,0,0,0,,0,,0,0,0,0,UP,1,1,0,0,0,89,0,,1,2,2,,0,,2,0,,0,L7OK,200,3,,,,,,,0,,,,0,0,,,,,-1,HTTP status check returned code <200>,,0,0,0,0,  # noqa
+            # postgres,postgresql_192.0.2.48_5432,0,0,0,0,100,0,0,0,,0,,0,0,0,0,DOWN,1,1,0,1,1,87,87,,1,2,3,,0,,2,0,,0,L7STS,503,2,,,,,,,0,,,,0,0,,,,,-1,HTTP status check returned code <503>,,0,0,0,0,  # noqa
+            # postgres,BACKEND,0,0,0,0,200,0,0,0,0,0,,0,0,0,0,UP,1,1,0,,0,89,0,,1,2,0,,0,,1,0,,0,,,,,,,,,,,,,,0,0,0,0,0,0,-1,,,0,0,0,0,  # noqa
+            haproxy_data = list(csv.reader(
+                haproxy_csv.lstrip('# ').splitlines()
+            ))
+            headers = haproxy_data[0]
+            haproxy_status = [
+                {
+                    headers[i]: row[i]
+                    for i in range(len(headers))
+                }
+                for row in haproxy_data
+            ]
+
+            servers = [
+                row for row in haproxy_status
+                if row['svname'] not in ('BACKEND', 'FRONTEND')
+            ]
+
+            for server in servers:
+                logger.debug(
+                    'Server: {name}: {status} ({why}) - {detail}'.format(
+                        name=server['svname'],
+                        status=server['status'],
+                        why=server['check_status'],
+                        detail=server['last_chk'],
+                    )
+                )
 
-    def _remove_files(self):
-        """
-        Remove all files related to the REST service and uninstall the RPM,
-        """
-        yum_remove('cloudify-rest-service')
-        yum_remove('cloudify-agents')
+            if any(server['check_status'] == 'INI' for server in servers):
+                logger.info('DB healthchecks still initialising...')
+                time.sleep(1)
+                continue
 
-        common.remove('/opt/manager')
+            healthy = True
+            # If we got here, haproxy is happy!
+            break
+
+        if not healthy:
+            raise RuntimeError(
+                'DB proxy startup failed.'
+            )
+
+        logger.info('DB proxy startup complete.')
+
+    def _set_haproxy_connect_any(self, enable):
+        """Make SELinux allow/disallow haproxy listening on 5432."""
```

Comment:
this allows connecting to any port, not listening on 5432 specifically, so the docstring should tell the truth

Reply:
Fair point. I'll update it.